### PR TITLE
matcher: improve parser

### PR DESCRIPTION
**Pull request by drjova**
* Improves `page_start` when reference has only one page.

* Fixes wrong config.

### CODE REVIEW

File: backend/inspirehep/matcher/config.py [hunk 1]
```diff
@@ -119,19 +119,6 @@
                     ],
                     "type": "nested",
                 },
-                {
```
Comment (drjova) [inline backend/inspirehep/matcher/config.py:122]: my bad, I was playing with the config and this was a left over. Now is the same with https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/refextract/config.py#L92

File: backend/inspirehep/matcher/parsers.py [hunk 2]
```diff
@@ -68,9 +68,14 @@ def journal_title(self):
...
     @property
     def page_start(self):
-        return self.root.xpath(
+        page_start = self.root.xpath(
             "/biblStruct/monogr/imprint/biblScope[@unit='page']/@from"
         ).get()
+        if page_start:
+            return page_start
+        return self.root.xpath(
+            "/biblStruct/monogr/imprint/biblScope[@unit='page']/text()"
```
Comment (michamos) [inline backend/inspirehep/matcher/parsers.py:77]: alternatively you can select both at the XPath level (only one of those should exist, and you're picking that one with `.get()`):
```suggestion
        page_start = self.root.xpath(
            "/biblStruct/monogr/imprint/biblScope[@unit='page']/(@from | text())"
        ).get()
```

Comment (drjova) [reply backend/inspirehep/matcher/parsers.py:77]: @michamos nice one!

Comment (drjova) [reply backend/inspirehep/matcher/parsers.py:77]: though: `ValueError: XPath error: Invalid expression in /biblStruct/monogr/imprint/biblScope[@unit='page']/(@from | text())`

Comment (drjova) [reply backend/inspirehep/matcher/parsers.py]: `(/biblStruct/monogr/imprint/biblScope[@unit='page']/@from | /biblStruct/monogr/imprint/biblScope[@unit='page']/text())` this seems ok

Comment (michamos) [reply backend/inspirehep/matcher/parsers.py]: strange, here it works: https://www.freeformatter.com/xpath-tester.html

**Pull request merged**